### ISSUE TITLE
WearOS Version/Mode

### ISSUE THREAD

**Issue by BeigeAnimal** [jmshrv/finamp#588]
I like this app and I use it on my Galaxy Watch 3 because I work in a location where I'm not allowed to bring my phone, I was wondering if it'd be possible to create a version for WearOS. 
The Samsung wearables has issues for me when syncing music which is why I'm looking for alternatives.
The app works pretty well as is the main issue is since it's a circular display things in the corners get cut off and it's impossible to select certain options, otherwise I'd just cope and deal with it.

**Comment by Chaphasilor** [comment]
I'm planning to add WearOS support down the road, but not before I get a new smartwatch (since my last one died). I wasn't sure if I wanted to try and make the whole app standalone (no connection to the phone app required) or not, but I guess it might be a good idea then. It's probably also easier to pull off in Flutter this way.  

In theory, all features should work just fine on a watch, including downloads. Just the interface needs to be adapted.

**Comment by TheRealGramdalf** [comment]
I have a samsung galaxy watch4 classic which runs WearOS 4, so I'm willing to do some beta testing if you like - integration with the bezel would also be very much appreciated.
In regards to downloads, I think standalone is absolutely the way to go - you can already control audio playing on your phone with a standard WearOS install, so the main benefit would be the ability to browse for songs in finamp on the watch and start it playing on the phone (as opposed to finding whatever song/playlist you want on the phone, and controlling play/pause/seek with the watch)

**Comment by Chaphasilor** [comment]
I actually managed to get this working on my old watch a few days ago:  
<img src="https://github.com/jmshrv/finamp/assets/18015147/cdf19b28-9a35-4ce2-b322-13c58075cb17" height="400">  
(ignore the dirty watchband please, it has been collecting dust for a few months since the battery is basically dead)

All features seem to be working, including playback and downloads:  
![image](https://github.com/jmshrv/finamp/assets/18015147/7164e810-d39e-4888-a8f6-1dab8b0f72a6)

But as you can see, the interface is absolutely unusable.  
We'll probably have to build a complete alternative UI for watches, that omits some of less-useful the features in the context of a wearable.

You can give it a try by using my branch at https://github.com/Chaphasilor/finamp/tree/wear-os. Just connect to your watch using ADB wireless debugging.  
I'd especially be interested in startup performance, since it takes 10-30 seconds on my watch, but that has a pretty outdated processor...

**Comment by ryanwwest** [comment]
I'd love this feature too.

**Comment by Felitendo** [comment]
Any updates?

**Comment by Chaphasilor** [comment]
Nope, my watch has died completely since then. I might get a Pixel Watch 3 when the price drops, but until then I won't be working on this.
